File: 1122-relative-sort-array/1122-relative-sort-array.java
class Solution {
    
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int j=0;

        int[] freq=new int[100000];
        int[] result=new int[arr1.length];

        for(int i:arr1){
            freq[i]++;
        }

        for(int i:arr2){
            while(freq[i]!=0){
                result[j++]=i;
                freq[i]--;
            }
        }

        for(int i=0;i<freq.length;i++){
            while(freq[i]!=0){
                result[j++]=i;
                freq[i]--;
            }
        }

        return result;
    }

}

File: 1122-relative-sort-array/1122-relative-sort-array.kt
class Solution {
fun relativeSortArray(arr1: IntArray, arr2: IntArray): IntArray {
        val map = HashMap<Int, Int>()
        arr1.forEach {
            map[it] = map.getOrDefault(it, 0) + 1
        }
        val result = IntArray(arr1.size)
        var index = 0
        arr2.forEach {
            val count = map[it]!!
            for (i in 0 until count) {
                result[index++] = it
            }
            map.remove(it)
        }
        val keys = map.keys.sorted()
        keys.forEach {
            val count = map[it]!!
            for (i in 0 until count) {
                result[index++] = it
            }
        }
        return result
    }
}